File: Semester 1/IN1000/Innlev 8/Conway's Game of Life/verden.py
from rutenett import Rutenett


class Verden:
    def __init__(self, rader, kolonner):
        self._rutenett = Rutenett(rader, kolonner)
        self._generasjonsnummer = 0

        self._rutenett.fyll_med_tilfeldige_celler()
        self._rutenett.koble_celler()

    def tegn(self):
        self._rutenett.tegn_rutenett()
        print(f"Generasjon: {self._generasjonsnummer} | Antall Levende Celler: {self._rutenett.antall_levende()}")

    def oppdatering(self):
        alle_celler = self._rutenett.hent_alle_celler()
        for celle in alle_celler:        # Til hver celle:
            celle.tell_levende_naboer()  # Teller levende naboer

        for celle in alle_celler:        # Til hver celle:
            celle.oppdater_status()      # Oppdaterer status

        self._generasjonsnummer += 1     # Oppdaterer generasjonsnummer

File: Semester 2/IN1010/Oblig 1/src/Verden.java
public class Verden {

    // Variables
    Rutenett rutenett;
    int genNr;

    // Constructor
    Verden(int antRader, int antKolonner) {
        this.rutenett = new Rutenett(antRader, antKolonner);  // Creating grid
        this.genNr = 0;

        this.rutenett.fyllMedTilfeldigeCeller();  // Filling with random cells
        this.rutenett.kobleAlleCeller();          // Connecting...
    }

    // Methods
    void tegn() {
        this.rutenett.tegnRutenett();  // Drawing...
        System.out.println("Generasjon: " + this.genNr);
        System.out.println("Antall levende celler: " + this.rutenett.antallLevende());
    }

    void oppdatering() {
        for (Celle[] rad : this.rutenett.rutene) {
            for (Celle celle : rad) {
                celle.tellLevendeNaboer();  // Updating each cell
                celle.oppdaterStatus();     // Including status and living neighbour amount
            }
        }
        this.genNr++;
        this.tegn();  // Drawing too...
    }
}
